trin-beacon/src/validation.rs

```diff
@@ -135,33 +139,83 @@ impl Validator<BeaconContentKey> for BeaconValidator {
             }
             BeaconContentKey::HistoricalSummariesWithProof(key) => {
                 let fork_versioned_historical_summaries =
-                    ForkVersionedHistoricalSummariesWithProof::from_ssz_bytes(content).map_err(
-                        |err| {
-                            anyhow!(
-                                "Historical summaries with proof has invalid SSZ bytes: {:?}",
-                                err
-                            )
-                        },
-                    )?;
+                    Self::general_summaries_validation(content, key)?;
+
+                let latest_finalized_root = self
+                    .header_oracle
+                    .read()
+                    .await
+                    .get_finalized_state_root()
+                    .await?;
+
+                Self::state_summaries_validation(
+                    fork_versioned_historical_summaries,
+                    latest_finalized_root,
+                )
+                .await?;
+            }
+        }
 
-                // Check if the historical summaries with proof epoch matches the content key epoch
-                if fork_versioned_historical_summaries
-                    .historical_summaries_with_proof
-                    .epoch
-                    != key.epoch
-                {
-                    return Err(anyhow!(
+        Ok(ValidationResult::new(true))
+    }
+}
+
+impl BeaconValidator {
+    /// General validation for the historical summaries with proof content
+    fn general_summaries_validation(
+        content: &[u8],
+        key: &HistoricalSummariesWithProofKey,
+    ) -> anyhow::Result<ForkVersionedHistoricalSummariesWithProof> {
+        let fork_versioned_historical_summaries =
+            ForkVersionedHistoricalSummariesWithProof::from_ssz_bytes(content).map_err(|err| {
+                anyhow!(
+                    "Historical summaries with proof has invalid SSZ bytes: {:?}",
```

Comment:
*nitpick* you can inline the `err` here, though this should be caught in the clippy check [here](https://github.com/ethereum/trin/blob/master/trin-beacon/src/lib.rs#L2) ... not sure why it's not being picked up?